cpp/src/arrow/builder-benchmark.cc

```diff
@@ -29,238 +29,212 @@
-  std::vector<uint64_t> data(size);
-  for (uint64_t i = 0; i < static_cast<uint64_t>(size); i++) {
-    data[i] = i;
-  }
-  while (state.KeepRunning()) {
-    AdaptiveUIntBuilder builder;
-    for (int64_t i = 0; i < size; i += chunk_size) {
-      // Build up an array of 128 MiB in size
-      ABORT_NOT_OK(builder.AppendValues(data.data() + i, chunk_size, nullptr));
+    for (int i = 0; i < kFinalSize; i++) {
+      ABORT_NOT_OK(builder.AppendValues(kData.data(), kData.size(), nullptr));
     }
+
     std::shared_ptr<Array> out;
     ABORT_NOT_OK(builder.Finish(&out));
   }
-  state.SetBytesProcessed(state.iterations() * data.size() * sizeof(int64_t));
+
+  state.SetBytesProcessed(state.iterations() * kBytesProcessed);
 }
 
-static void BM_BuildAdaptiveUIntNoNullsScalarAppend(
+static void BuildAdaptiveIntNoNullsScalarAppend(
     benchmark::State& state) {  // NOLINT non-const reference
-  int64_t size = static_cast<int64_t>(std::numeric_limits<int16_t>::max()) * 256;
-  std::vector<uint64_t> data(size);
-  for (uint64_t i = 0; i < static_cast<uint64_t>(size); i++) {
-    data[i] = i;
-  }
-  while (state.KeepRunning()) {
-    AdaptiveUIntBuilder builder;
-    for (int64_t i = 0; i < size; i++) {
-      ABORT_NOT_OK(builder.Append(data[i]));
+  for (auto _ : state) {
+    AdaptiveIntBuilder builder;
+
+    for (int i = 0; i < kFinalSize; i++) {
+      for (size_t j = 0; j < kData.size(); j++) {
+        ABORT_NOT_OK(builder.Append(kData[i]))
+      }
     }
+
     std::shared_ptr<Array> out;
     ABORT_NOT_OK(builder.Finish(&out));
   }
-  state.SetBytesProcessed(state.iterations() * data.size() * sizeof(int64_t));
+
+  state.SetBytesProcessed(state.iterations() * kBytesProcessed);
 }
 
-static void BM_BuildBooleanArrayNoNulls(
+static void BuildBooleanArrayNoNulls(
     benchmark::State& state) {  // NOLINT non-const reference
-  // 2 MiB block
-  std::vector<uint8_t> data(2 * 1024 * 1024);
-  constexpr uint8_t bit_pattern = 0xcc;  // 0b11001100
-  uint64_t index = 0;
-  std::generate(data.begin(), data.end(),
-                [&]() -> uint8_t { return (bit_pattern >> ((index++) % 8)) & 1; });
-
-  while (state.KeepRunning()) {
+
+  size_t n_bytes = kData.size() * sizeof(ValueType);
+  const uint8_t* data = reinterpret_cast<const uint8_t*>(kData.data());
+
+  for (auto _ : state) {
     BooleanBuilder builder;
+
     for (int i = 0; i < kFinalSize; i++) {
-      // Build up an array of 512 MiB in size
-      ABORT_NOT_OK(builder.AppendValues(data.data(), data.size()));
+      ABORT_NOT_OK(builder.AppendValues(data, n_bytes));
     }
+
     std::shared_ptr<Array> out;
     ABORT_NOT_OK(builder.Finish(&out));
   }
-  state.SetBytesProcessed(state.iterations() * data.size() * kFinalSize);
-}
 
-static void BM_BuildBinaryArray(benchmark::State& state) {  // NOLINT non-const reference
-  // About 160MB
-  const int64_t iterations = 1 << 24;
-  std::string value = "1234567890";
+  state.SetBytesProcessed(state.iterations() * kBytesProcessed);
```

Comment:
I find `kBytesProcessPerRound` and `kBytesProcessed` a bit confusing, at first `round` and `iteration` seem synonyms, but they mean different loops.

Reply:
Correct, `iterations` is a google benchmark concept. Round is used to make a distinction between batch APIs versus point APIs.

Reply:
I'll rename `kFinalSize` to `kRounds` for this.